### PR TITLE
[NO GBP] Healing Virus Selfcure Hotfix

### PR DESCRIPTION
**Pull request by Higgin**
## About The Pull Request

Fixes a logic issue in how healing viruses were supposed to self-cure that basically never came out in practice but wasn't intended. 

## Why It's Good For The Game

fix good, makes it so starving nutrition OR negative satiety can mess up a healing virus, makes TRAIT_NOHUNGER properly protect against losing a healing virus without spaceacillin

## Changelog

:cl:
fix: Healing viruses now no longer self-cure for reasons they're not supposed to and do for those that they are.
/:cl:

### CODE REVIEW

**Comment by san7890** [comment]
brother that is one ugly commit history you should really base new branches off `tgstation/tgstation:master` instead of your old branches

File: code/datums/diseases/_disease.dm [hunk 1]
```diff
@@ -105,7 +105,7 @@
 		switch(severity)
 			if(DISEASE_SEVERITY_POSITIVE) //good viruses don't go anywhere after hitting max stage - you can try to get rid of them by sleeping earlier
 				cycles_to_beat = max(DISEASE_RECOVERY_SCALING, DISEASE_CYCLES_POSITIVE) //because of the way we later check for recovery_prob, we need to floor this at least equal to the scaling to avoid infinitely getting less likely to cure
-				if((HAS_TRAIT(affected_mob, TRAIT_NOHUNGER)) || affected_mob.nutrition < NUTRITION_LEVEL_STARVING || affected_mob.satiety < 0 || slowdown == 1) //any sort of malnourishment/immunosuppressant opens you to losing a good virus
+				if((HAS_TRAIT(affected_mob, TRAIT_NOHUNGER)) || ((affected_mob.nutrition > NUTRITION_LEVEL_STARVING) && (affected_mob.satiety >= 0)) && slowdown == 1) //any sort of malnourishment/immunosuppressant opens you to losing a good virus
```
Comment (san7890) [inline code/datums/diseases/_disease.dm:108]:
```suggestion
				if(HAS_TRAIT(affected_mob, TRAIT_NOHUNGER) || ((affected_mob.nutrition > NUTRITION_LEVEL_STARVING) && (affected_mob.satiety >= 0) && (slowdown == 1))) //any sort of malnourishment/immunosuppressant opens you to losing a good virus
```

this if() check makes zero sense in its current state and is confusing as hell to read, is this what you meant? either we have TRAIT_NOHUNGER _or_ we meet all three criteria to being malnourished? if this isn't what you intended, really consider where you're placing those parenthesis.

**Review by san7890 — CHANGES_REQUESTED** [review]

**Comment by Higgin** [comment]
> brother that is one ugly commit history you should really base new branches off `tgstation/tgstation:master` instead of your old branches - will help in case you get a dirty branch or have changes that you don't want showing up in your PR diffs.

yeah and I'm not sure how to squash all those commits like I used to or would if it was all in one PR now - any advice for doing that through VSC or tortoisegit would be appreciated at this point unless it would just be cleaner to reopen a new branch off the master to avoid this mess

File: code/datums/diseases/_disease.dm (hunk 1 above)
Comment (Higgin) [reply code/datums/diseases/_disease.dm:108]: slowdown isn't a criterion for malnourishment, it's spaceacillin/TRAIT_VIRUS_RESIST

what I'm trying to get at and maybe would be better broken out of one line is that 

if we have trait_nohunger, malnutrition/ poor satiety doesn't matter
but if we have trait_nohunger, slowdown being something other than 1 (unmodified) still does

Comment (Higgin) [reply code/datums/diseases/_disease.dm:108]: image
if it helps at all, this is what I'm working with in vsc

Comment (Higgin) [reply code/datums/diseases/_disease.dm]: If anything, I'm realizing that's now also wrong for how it interacts with NOHUNGER, and it should probably instead be this - even though what I did fixed the problem with nutrition for regular mobs, I missed how that was interacting with values that aren't really supposed to be there (nutrition/satiety with ethereals for instance.)

image

**Pull request merged**